compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -533,11 +533,11 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         layout: Option<TyAndLayout<'tcx>>,
     ) -> InterpResult<'tcx, OpTy<'tcx, M::Provenance>> {
         use rustc_middle::mir::Operand::*;
-        let op = match *mir_op {
+        let op = match mir_op {
             // FIXME: do some more logic on `move` to invalidate the old location
-            Copy(place) | Move(place) => self.eval_place_to_op(place, layout)?,
+            &(Copy(place) | Move(place)) => self.eval_place_to_op(place, layout)?,
```

Comment:
Not sure if this is an improvement here, maybe
```suggestion
            Copy(place) | Move(place) => self.eval_place_to_op(*place, *layout)?,
```
would be even nicer?

Reply:
I don't think derefing here is nicer. Another alternative is
```suggestion
            &Copy(place) | &Move(place) => self.eval_place_to_op(place, layout)?,
```